### PR TITLE
Delegate window

### LINKED ISSUE #28: [2] Remove spinning triangle
As a user, when I first open the app, I would like the OpenGL area to be empty, and contain a shortcut to "Open a model".  Once a model is loaded, this shortcut will be hidden.

### LINKED ISSUE #29: [2] Add loading screen
As a user, I expect to see a loading bar or "loading..." text while the model is being loaded.  If possible, move the loading to a separate thread so the UI is still functional.

### PR DESCRIPTION
**Pull request by tigerman9854**
Closes #28
Closes #29

- Added a new model, low poly tree
- Deleted spinning triangle
- Added 4 states of our app (see screenshots below)
    1. Welcome
    2. Viewing Model
    3. Loading
    4. Error
- Changed FOV to 45
- Added unit tests for different modes
- Added ability to close a file

### CODE REVIEW

**Comment by tigerman9854** [comment]
image

**Comment by tigerman9854** [comment]
image
image
image
image
image

File: ModelViewer/ModelViewer/GraphicsWindowDelegate.cpp [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+	connect(m_pGraphicsWindow, &ViewerGraphicsWindow::BeginModelLoading, this, &GraphicsWindowDelegate::OnBeginModelLoading);
+	connect(m_pGraphicsWindow, &ViewerGraphicsWindow::EndModelLoading, this, &GraphicsWindowDelegate::OnEndModelLoading);
+	connect(m_pGraphicsWindow, &ViewerGraphicsWindow::Error, this, &GraphicsWindowDelegate::OnError);
+	connect(m_pGraphicsWindow, &ViewerGraphicsWindow::ModelUnloaded, this, &GraphicsWindowDelegate::OnModelUnloaded);
+}
+
+void GraphicsWindowDelegate::SetStatus(Status s)
+{
+	m_status = s;
+	DoOnStatusChanged();
```
Comment (richjoregonstate) [inline ModelViewer/ModelViewer/GraphicsWindowDelegate.cpp:90]: This is a cool design pattern :)

**Review by richjoregonstate — APPROVED** [review]
Looks great!

File: ModelViewer/ModelViewer/GraphicsWindowDelegate.cpp (hunk 1 above)
Comment (tigerman9854) [reply ModelViewer/ModelViewer/GraphicsWindowDelegate.cpp:90]: Thanks, it's event based design... to the extreme!

**Pull request merged**